frontend/src/component/AddVisitCountry.tsx

```diff
@@ -1,43 +1,100 @@
+import {SyntheticEvent, useState} from "react";
 import {Traveller} from "../model/User";
 import {useNavigate} from "react-router-dom";
 import {updateUser} from "../apiCalls";
-import {Autocomplete, TextField} from "@mui/material";
+import {Autocomplete, Box, Button, Chip, LinearProgress, TextField, Typography} from "@mui/material";
+import useCountries from "../hook/useCountries";
+import {AddCircle} from "@mui/icons-material";
 
 type AddVisitCountryProps = {
-    countries: Country[],
     loggedInTraveller: Traveller
 }
-export default function AddVisitCountry(props: AddVisitCountryProps) {
+export default function AddVisitedCountry(props: AddVisitCountryProps) {
     const navigate = useNavigate()
+    const {countries} = useCountries()
+    const [selectedCountries, setSelectedCountries] = useState<Country[]>([])
 
-    function addCountryToUser(event: SyntheticEvent<Element, Event>, value: Country | null) {
+    function addCountriesToUser() {
         if (!props.loggedInTraveller.visitedCountries) {
             props.loggedInTraveller.visitedCountries = []
         }
+        const newPlusOldCountries = [...props.loggedInTraveller.visitedCountries, ...selectedCountries]
+        const copyLoggedInTraveller = props.loggedInTraveller
+        copyLoggedInTraveller.visitedCountries = newPlusOldCountries
+
+        updateUser(props.loggedInTraveller.id, copyLoggedInTraveller)
+            .then(() => {
+                props.loggedInTraveller.visitedCountries = newPlusOldCountries
+                navigate("/overview")
+            })
+            .catch(console.error)
+    }
+
+    function handleRemoveCountryFromList(countryId: string) {
+
+        setSelectedCountries((prevState) => {
+            return prevState.filter((country) => country.threeLetterCode !== countryId);
+        })
+    }
+
+    function addCountryToList(event: SyntheticEvent<Element, Event>, value: Country | null) {
         if (value) {
-            const copyVisitedCountries = [...props.loggedInTraveller.visitedCountries,value]
-            const copyLoggedInTraveller = props.loggedInTraveller
-            copyLoggedInTraveller.visitedCountries = copyVisitedCountries
-
-            updateUser(props.loggedInTraveller.id, copyLoggedInTraveller)
-                .then(() => {
-                    props.loggedInTraveller.visitedCountries= copyVisitedCountries
-                    navigate("/overview")
-                })
-                .catch(console.error)
+            setSelectedCountries((prevState) => {
+                return [...prevState, value]
+            })
         }
+    }
+
+    const buttonDisabled: boolean = selectedCountries.length < 1
+
+    if (countries) {
+        return (<Box display={"flex"}
+                     width={'100%'}
+                     justifyContent={"center"}
+                     marginTop={20}
+                     flexDirection={"column"}
+                     flexWrap={"wrap"}
+                     alignItems={"center"}
```

Comment:
Personally not a fan of this approach, since it appears to be a little less reusable to my liking.
You can absolutely use this way though, however, please be consistent across your code base (and I've seen a lot of classes so far :smiley: )

Reply:
Slowly and steady changing it to that approach